packages/protocol-kit/src/Safe.ts

```diff
@@ -421,13 +455,37 @@ class Safe {
     return this.#moduleManager.isModuleEnabled(moduleAddress)
   }
 
+  /**
+   * Checks if a specific address or passkey is an owner of the current Safe.
+   *
+   * @param owner - The owner address or a passkey object
+   * @returns TRUE if the account is an owner
+   */
+  async isOwner(owner: string | passkeyArgType): Promise<boolean> {
+    const isOwnerAddress = typeof owner === 'string'
+
+    if (isOwnerAddress) {
+      return this.#isOwnerAddress(owner)
+    }
+
+    // passkey flow
+    const webAuthnSignerFactoryContract = this.#contractManager.safeWebAuthnSignerFactoryContract
+    const provider = this.#safeProvider.getExternalProvider()
+
+    const passkeySigner = await PasskeySigner.init(owner, webAuthnSignerFactoryContract, provider)
+
+    const ownerAddress = await passkeySigner.getAddress()
```

Comment:
I may be wrong but isnt all this chunk just the `this.#safeProvider.getSignerAddress()`?

Reply:
the `this.#safeProvider` might have been initialized with an EOA instead of a passkey. We need to ensure that the owner's address is created from the passkey provided by the user in the `isOwner` function parameter, not from the one in the `this.#safeProvider`.